src/hooks/useClickOutside.ts

```diff
@@ -0,0 +1,26 @@
+import { RefObject, useCallback, useEffect } from "react";
+
+export const useClickOutside = (
+  ref: RefObject<Element>,
+  callback: () => void,
+) => {
+  const handleClickOutside = useCallback(
```

Comment:
I don't se the point in this `useCallback`, since you create new instance of `callback` on each render of map component

Reply:
Without this I get a warning:

The 'handleClickOutside' function makes the dependencies of useEffect Hook (at line 19) change on every render. Move it inside the useEffect callback. Alternatively, wrap the definition of 'handleClickOutside' in its own useCallback() Hook.eslint[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)

I'll put it inside useEffect then.